src/Compilers/Core/VBCSCompiler/CSharpCompilerServer.cs

```diff
@@ -15,26 +17,27 @@ internal CSharpCompilerServer(string[] args, string clientDirectory, string base
         {
         }
 
-        public static int RunCompiler(
+        public static BuildResponse RunCompiler(
             string clientDirectory,
             string[] args,
             string baseDirectory,
             string sdkDirectory,
             string libDirectory,
             IAnalyzerAssemblyLoader analyzerLoader,
-            TextWriter output,
-            CancellationToken cancellationToken,
-            out bool utf8output)
+            CancellationToken cancellationToken)
         {
             var compiler = new CSharpCompilerServer(args, clientDirectory, baseDirectory, sdkDirectory, libDirectory, analyzerLoader);
-            utf8output = compiler.Arguments.Utf8Output;
+            bool utf8output = compiler.Arguments.Utf8Output;
 
-            foreach (var analyzer in compiler.Arguments.AnalyzerReferences)
+            if (!AnalyzerConsistencyChecker.Check(baseDirectory, compiler.Arguments.AnalyzerReferences, analyzerLoader))
```

Comment:
Minor: So we will be loading all the analyzer assemblies (and dependencies) even before we get to parsing the command line options. Will the compiler logo also be displayed during the subsequent `compiler.Run` invocation below? If so, can this cause a perceived delay to user between triggering a build and seeing that the build has started (when there are many analyzer references)?